metals/src/main/scala/scala/meta/metals/MetalsServices.scala

```diff
@@ -411,7 +413,25 @@ class MetalsServices(
       response
     case SbtConnect =>
       Task {
-        connectToSbtServer()
+        val path = cwd.resolve("project").resolve("build.properties")
+        val input = Files.newInputStream(path.toNIO)
+        val sbtVersion: Option[String] =
+          Try {
+            val props = new Properties()
+            props.load(input)
+            Option(props.getProperty("sbt.version"))
```

Comment:
is this `Option.apply` necessary inside a `Try.apply`?

Reply:
Yes, `Try` is for a possible failure in `props.load` and `Option` is for a possible `null` from `getProperty`, but I just realised that I should rewrite it in a simpler way.